### PR TITLE
feat: Add get util

### PR DESCRIPTION
**Pull request by zevaryx**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition

## Description

Discord.py had 2 utilities for searching an iterable, `find` and `get`. Dis-Snek already implements `find`, but is missing an implementation for `get`. This PR adds that missing implementation.

## Changes

- Add `dis_snek.client.misc_utils.get`

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

**Review by silasary — APPROVED** [review]

**Comment by leestarb** [comment]
- [x] Looks good

**Review by leestarb — APPROVED** [review]

**Review by benwoo1110 — APPROVED** [review]
looks good, maybe we should have a `get_all` or similar where it returns all elements that matches, not just the first one.

**Comment by zevaryx** [comment]
Honestly not a bad idea. Want me to add a get_all and find_all to this PR?

**Comment by silasary** [comment]
That would be great

**Review by leestarb — CHANGES_REQUESTED** [review]
For speedup purposes, please use the list literal instead of the function (C408)

File: dis_snek/client/utils/misc_utils.py [hunk 1]
```diff
@@ -45,6 +45,84 @@ def find(predicate: Callable, sequence: Iterable) -> Optional[Any]:
...
+
+    Returns:
+        A list of matches
+
+    """
+    matches = list()
+    for el in sequence:
+        if predicate(el):
+            matches.append(el)
+    return matches
```
Comment (leestarb) [inline dis_snek/client/utils/misc_utils.py]: You could rewrite this as list comprehension in a line.

**Comment by zevaryx** [comment]
> For speedup purposes, please use the list literal instead of the function (C408)

Changed

**Review by LordOfPolls — APPROVED** [review]

**Comment by LordOfPolls** [comment]
Pending re-review from @leestarb

**Review by leestarb — APPROVED** [review]
Ok, but

File: dis_snek/client/utils/misc_utils.py (hunk 1 above)
Comment (leestarb) [reply dis_snek/client/utils/misc_utils.py]:
```python
return [el for el in sequence if predicate(el)]
```

Comment (zevaryx) [reply dis_snek/client/utils/misc_utils.py]: Changed in latest push

**Review by benwoo1110 — APPROVED** [review]

**Review by leestarb — APPROVED** [review]

**Pull request merged**